spatial editor plugin bug fix

jeffuntildeath opened the pull request:
fix for crash when snapping a node to ground that has no
collision shape assigned

<!--
Pull requests should always be made for the `master` branch first, as that's
where development happens and the source of all future stable release branches.

Relevant fixes are cherry-picked for stable branches as needed.

Do not create a pull request for stable branches unless the change is already
available in the `master` branch and it cannot be easily cherry-picked.
Alternatively, if the change is only relevant for that branch (e.g. rendering
fixes for the 3.2 branch).
-->

jeffuntildeath committed:
spatial editor plugin bug fix

fix for crash when snapping a node to ground that has no
collision shape assigned

aaronfranke commented on editor/plugins/spatial_editor_plugin.cpp:
```diff
@@ -5798,21 +5798,31 @@ void SpatialEditor::snap_selected_nodes_to_floor() {
 		Spatial *sp = Object::cast_to<Spatial>(E->get());
 		if (sp) {
-			Vector3 from = Vector3();
+			Vector3 from = sp->get_global_transform().origin;
```
As this code is now, this value will be discarded if `cs.size()` and `found_valid_shape`, or if `cs.size()` is false and `vi.size()`. This wasn't the case before.

jeffuntildeath replied:
oops! correct, it is called snap_selected_nodes_to_floor(), not ground. I created an issue with a sample project here: https://github.com/godotengine/godot/issues/44231

I didn't see the code in the master branch at first since it changed source files. here is the PR for that:
https://github.com/godotengine/godot/pull/44234

hope that helps!

aaronfranke reviewed (commented):
What do you mean by "ground"? Can you provide a minimal reproduction project that shows the bug?

Pull requests should always be made for the `master` branch first, as that's where development happens and the source of all future stable release branches. Relevant fixes are cherry-picked for stable branches as needed.

jeffuntildeath committed:
fix for snap to floor editor crash bug

fix for issue #44231: snap_selected_nodes_to_floor() results in an editor
crash when a child collisionshape has invalid shape object

jeffuntildeath committed:
clang-format

Calinou commented:
Closing in favor of https://github.com/godotengine/godot/issues/44231, as https://github.com/godotengine/godot/issues/44231 is likely trivial to cherry-pick to the `3.2` branch.
